osmnx/geometries.py

```diff
@@ -407,8 +409,16 @@ def _create_gdf(response_jsons, polygon, tags):
     # Set default crs
     GDF.crs = settings.default_crs
 
+    # Apply .buffer(0) to any invalid geometries
+    GDF = _buffer_invalid_geometries(GDF)
+
     # Filter final gdf to requested tags and bounding polygon
-    GDF = _filter_final_gdf(gdf=GDF, polygon=polygon, tags=tags)
+    GDF = _filter_gdf_by_polygon_and_tags(GDF, polygon=polygon, tags=tags)
+
+    # bug in geopandas <=0.8.1 raises a TypeError if trying to plot empty geometries
+    # but missing geometries (gdf['geometry'] = None) cannot be projected e.g. gdf.to_crs()
+    # don't see any option other than to drop rows with missing/empty geometries
+    GDF = GDF[~(GDF["geometry"].isna() | GDF["geometry"].is_empty)].copy()
```

Comment:
What version of geopandas fixes this bug? Is it forthcoming in a future release?

Reply:
I raised these two issues last week, they've flagged as targeted for 0.9.

[BUG: Plotting empty geometries generates a Type Error (shapely only) #1572](https://github.com/geopandas/geopandas/issues/1572)
[BUG: to_crs() fails if there's None in geometry #1573](https://github.com/geopandas/geopandas/issues/1573)